### PR TITLE
plot.beachball: added fuzzy beachball

### PR DESCRIPTION
**Pull request by hvasbath**
Function for plotting a fuzzy beachball (beachball with uncertainties)
With and without best fit solution
fuzzy_bb_max
fuzzy_bb_no

### CODE REVIEW

**Comment by hvasbath** [comment]
This does not work for more than 500 MTs. As the alpha number precision in matplotlib is not fine enough to handle finer. Its on their todo list.
https://github.com/matplotlib/matplotlib/issues/3371
https://github.com/matplotlib/matplotlib/issues/4322

Any other suggestions to fix? Could we get some kind of number array out of the beachball plot to stack them on a raster image first, then do the normalization and then the plotting? @emolch

**Comment by hvasbath** [comment]
ah just found the pixmap implementation, awesome ...

**Comment by hvasbath** [comment]
Wow that was worth it. Now it actually the true fuzzy bb. The previous doesnt show all the details. Doing it with alpha wont work. The pixmap version works now. Saving as pdf would work, but the image would not be as on the screen still. png works. 
Also the issue with many mts is resolved now infinite amount of mts is working.
fuzzy_bb_max
fuzzy_bb_no

**Comment by hvasbath** [comment]
so if no complaints come, I will merge it? @miili @emolch

File: src/plot/beachball.py [hunk 1]
```diff
@@ -566,6 +567,138 @@ def plot_beachball_mpl(
...
+
+    x = num.linspace(-1., 1., nx)
+    y = num.linspace(-1., 1., ny)
+
+    vecs2 = num.zeros((nx * ny, 2), dtype=num.float)
+    vecs2[:, 0] = num.tile(x, ny)
+    vecs2[:, 1] = num.repeat(y, nx)
+
+    ii_ok = vecs2[:, 0]**2 + vecs2[:, 1]**2 <= 1.0
+    amps = num.zeros(nx * ny, dtype=num.float)
```
Comment (miili) [inline src/plot/beachball.py]: Numpy brings this signature 
``` python
amps = num.full(nx * ny, num.nan, dtype=num.float)
```

File: src/plot/beachball.py [hunk 2]
```diff
@@ -566,6 +567,138 @@ def plot_beachball_mpl(
...
+        array or sequence which can be converted into an MT object
+    :param best_mt: :py:class:`pyrocko.moment_tensor.MomentTensor` object or
+        an array or sequence which can be converted into an MT object
+        of most likely or minimum misfit solution to extra highlight
+    :param best_color: mpl color for best MomentTensor edges,
+        polygons are not plotted
+
+    See plot_beachball_mpl for other arguments
+
+    Note: The related axes should only be saved as raster image (e.g. png)
```
Comment (miili) [inline src/plot/beachball.py]: use
```
.. note :: 
```
or
```
.. warning ::
```

**Comment by hvasbath** [comment]
Great will do that!

**Comment by hvasbath** [comment]
can you take it from here? what is the dev? an extra branch? would be good to announce that somewhere if we differ from the previous commit/PR scheme

**Comment by miili** [comment]
> can you take it from here? what is the dev? an extra branch? would be good to announce that somewhere if we differ from the previous commit/PR scheme

Will do!

File: src/plot/beachball.py [hunk 3]
```diff
@@ -566,6 +567,138 @@ def plot_beachball_mpl(
...
+    # draw optimum edges
+    if best_mt is not None:
+        best_amps, bx, by = mts2amps(
+            [best_mt],
+            grid_resolution=grid_resolution,
+            projection=projection,
+            beachball_type=beachball_type,
+            mask=False)
+
+        axes.contour(
```
Comment (emolch) [inline src/plot/beachball.py:681]: Did you try how it looks like when you use one of the pcolor variants rather than contourf?

Comment (hvasbath) [reply src/plot/beachball.py:681]: no I dont know that function, will have to try

Comment (hvasbath) [reply src/plot/beachball.py:681]: Have done it now @emolch looks not as nice, because you see the grid of the matrix, the contours are way better

**Pull request merged**